packages/flutter/lib/src/cupertino/button.dart

```diff
@@ -0,0 +1,113 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+const TextStyle _kCupertinoButtonStyle = const TextStyle(
+  fontFamily: '.SF UI Text',
+  inherit: false,
+  fontSize: 14.0,
+  fontWeight: FontWeight.normal,
+  color: const Color(0xFF007AFF),
+  textBaseline: TextBaseline.alphabetic,
+);
+
+final TextStyle _kCupertinoDisabledButtonStyle = _kCupertinoButtonStyle.copyWith(
+  color: const Color(0xFFC4C4C4),
+);
+
+/// An iOS style button.
+///
+/// Takes in a text or an icon that fades out and in on touch. May optionally have an
+/// invariant border.
+///
+/// See also:
+///
+///  * <https://developer.apple.com/ios/human-interface-guidelines/ui-controls/buttons/>
+class CupertinoButton extends StatefulWidget {
+  CupertinoButton({
+    this.text,
+    this.padding: 16.0,
+    @required this.onPressed,
+  });
+
+  final String text;
```

Comment:
Rather than taking text, this should take a child widget.  The client can supply a Text widget if they want, but they can also supply something else more interesting.